kedro-org/kedro-viz
# Adjust requirements file and dependabot versioning strategy

## Description

Resolves #1967 

## Development notes

- Following the doc mentioned [here](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#versioning-strategy) , the dependabot versioning-strategy `widen` would include both the new and old versions when possible, allowing users the flexiblility to install Kedro-Viz. Unfortunately, the `pip` ecosystem does not support `widen`. We will use `auto` (which is widen for libraries)
- Relaxed the allowed versions in `requirements.txt` file to have a range with upper-bound corresponding to a major release of the package (which corresponds to a non-backward compatible change, hence needs more attention when upgrading).

## QA notes

- No change in the app behavior
- All tests should pass
- Try installing Kedro-Viz and prefect>=3.0.0rc10. This should not raise any issues as mentioned in #1967 . Note that there is an issue with watchgod not having support for anyio >= 4 and prefect >=3.0.0rc10 requires anyio >= 4. There is an open [discussion](https://github.com/samuelcolvin/watchfiles/issues/254) in watchgod.

## Checklist
